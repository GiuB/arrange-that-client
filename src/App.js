import React, { Component } from 'react'
import './App.css'
import router from 'config/router'
import { Provider } from 'react-redux'
import store from 'store/index'
import { GoogleLogin } from 'react-google-login'
import config from './config.json'

class App extends Component {

    constructor() {
        super();
        this.state = { isAuthenticated: false, user: null, token:
            ''};
    }
    logout = () => {
        this.setState({isAuthenticated: false, token: '', user: null})
    };

    onFailure = (error) => {
        alert(error);
    };

    googleResponse = (response) => {
        console.log(response);
        const tokenBlob = new Blob([JSON.stringify({access_token: response.accessToken}, null, 2)], {type : 'application/json'});
        const options = {
            method: 'POST',
            body: tokenBlob,
            mode: 'cors',
            cache: 'default'
        };
        fetch('https://accounts.google.com/signin/oauth/oauthchooseaccount?client_id=206945578523-0h8t8i7k5d09j0vg31ncspa4pbrddff6.apps.googleusercontent.com&as=xFDBemLg1L9ik5llB8K7Uw&destination=http%3A%2F%2Farrangement-server.herokuapp.com&approval_state=!ChRSWklSU2I5OHRWZDJGMVVzUnJMVhIfQTlDMmZ0Y2VkaW9aMEs2dFEwd1Nsa3VsS1ZfSGRSWQ%E2%88%99APNbktkAAAAAXABpjOv6r9QOSQUANtEjaqia7gTv234E&oauthgdpr=1&xsrfsig=AHgIfE85dgKn7GfOJkXL2gDfCd67Z6reRA&flowName=GeneralOAuthFlow', options).then(r => {
            const token = r.headers.get('x-auth-token');
            r.json().then(user => {
                if (token) {
                    this.setState({isAuthenticated: true, user, token})
                }
            });
        })
    };

    render() {
        let content = !!this.state.isAuthenticated ?
            (
                <div>
                    <p>Authenticated</p>
                    <div>
                        {this.state.user.email}
                    </div>
                    <div>
                        <button onClick={this.logout} className="button">
                            Log out
                        </button>
                    </div>
                </div>
            ) :
            (
                <div>
                    <GoogleLogin
                        clientId={config.GOOGLE_CLIENT_ID}
                        buttonText="Login"
                        onSuccess={this.googleResponse}
                        onFailure={this.onFailure}
                    />
                </div>
            );

        return (
            <div className="App">
                {content}
            </div>
        );
    }
}

// TODO: finish implementing the Google SSO and add back the Route function 
//         Route information
//
//          render() {
//              return (
//                  <Provider store={store}>
//                      <div className="App">
//                          { router }
//                     // <Routes name={this.state.appName}/>
//                      </div>
//                  </Provider>
//              )
//          }
// }

export default App
