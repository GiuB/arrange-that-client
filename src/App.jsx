import React, { Component } from 'react'
import { Provider } from 'react-redux'
import router from 'config/router'
import store from 'store/store'
import { SnackbarProvider } from 'notistack';
import axios from 'axios';
import { logout } from 'actions/account/account'

import './App.css'

class App extends Component {
    render() {
        return (
            <SnackbarProvider 
                maxSnack={3} preventDuplicate 
                anchorOrigin={{
                    vertical: 'bottom',
                    horizontal: 'right',
                }}
            >
                <Provider store={store}>
                    <div className="App">
                        { router }
                    </div>
                </Provider>
            </SnackbarProvider>
        )
    }
}

/** Intercept any unauthorized request.
* dispatch logout action accordingly **/
const UNAUTHORIZED = 401;
const INTERNAL_SERVER_ERROR = 500;
const {dispatch} = store; // direct access to redux store.
axios.interceptors.response.use(
    response => response,
    error => {
        const {status} = error.response;
        if (status === UNAUTHORIZED || status === INTERNAL_SERVER_ERROR) {
            console.log("logging out due to timeout")
            dispatch(logout());
        }
        return Promise.reject(error);
    }
);

export default App