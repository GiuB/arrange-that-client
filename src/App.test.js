import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json'

import App from './App';

Enzyme.configure({adapter: new Adapter()});

it('renders without crashing', () => {
    const app = shallow(<App/>);

    expect(toJson(app)).toMatchSnapshot();
});
