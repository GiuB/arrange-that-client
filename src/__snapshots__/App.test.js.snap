// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<SnackbarProvider
  anchorOrigin={
    Object {
      "horizontal": "right",
      "vertical": "bottom",
    }
  }
  dense={false}
  maxSnack={3}
  preventDuplicate={true}
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <div
      className="App"
    >
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Default>
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <PrivateRoute
              component={[Function]}
              exact={true}
              path="/arrangement/:arrangement_id"
            />
            <PrivateRoute
              component={[Function]}
              exact={true}
              path="/all_arrangements"
            />
            <PrivateRoute
              component={[Function]}
              exact={true}
              path="/feedback"
            />
          </Switch>
        </Default>
      </Router>
    </div>
  </Provider>
</SnackbarProvider>
`;
